### PR TITLE
Dockerfile use the Makefile to build tikv

### PR DESCRIPTION
**Pull request by xiamengyu**
## What have you changed? (mandatory)
1. Add a new rule `build_release` to Makefile
2. Replace `RUN cargo build --no-default-features --release --features "jemalloc portable sse no-fail"` with `RUN make build_release` in script `scripts/gen-dockerfile.sh`  and execute the script to generate a new Dockerfile.

## What are the type of the changes? (mandatory)
- Improvement

## How has this PR been tested? (mandatory)
Run `docker build .` on my local MacOS.

## Does this PR affect documentation (docs) or release note? (mandatory)
No

## Does this PR affect tidb-ansible update? (mandatory)
No

## Refer to a related PR or issue link (optional)
[#4845](https://github.com/tikv/tikv/issues/4845)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -61,7 +61,7 @@ RUN mkdir -p ./src/bin && \
 COPY ./src ./src
 COPY ./components ./components
 
-RUN cargo build --no-default-features --release --features "jemalloc portable sse no-fail" 
+RUN make build
```
Comment (siddontang) [inline Dockerfile]: seem we don't copy the Makefile to the docker before this command.

Have you run to verify that it can work?

**Comment by siddontang** [comment]
thanks @xiamengyu

File: Dockerfile (hunk 1 above)
Comment (overvenus) [reply Dockerfile]: It builds a debug tikv, I think it's better to `make release`

Comment (siddontang) [reply Dockerfile]: Friendly ping @xiamengyu

Comment (xiamengyu) [reply Dockerfile]: @siddontang @overvenus Thank you for pointing out the problem, I just fixed it. Verified on my local MacOS. 
Please take another review.

**Pull request merged**